src/main/extension/extension_load.cpp

```diff
@@ -67,8 +67,15 @@ struct ExtensionAccess {
 	static void SetError(duckdb_extension_info info, const char *error) {
 		auto &load_state = DuckDBExtensionLoadState::Get(info);
 
-		load_state.has_error = true;
-		load_state.error_data = ErrorData(ExceptionType::UNKNOWN_TYPE, error);
+		if (error) {
+			load_state.has_error = true;
+			load_state.error_data = ErrorData(ExceptionType::UNKNOWN_TYPE, error);
```

Comment:
```suggestion
			load_state.error_data = ErrorData(error);
```

Avoiding the ExceptionType would allow extensions to pass the type via JSON-encoding.
Default is INVALID_TYPE, but that might be fine and still better given the flexibility offered.

Or maybe ErrorData should have an API to override if still INVALID, or something like that.